### PR TITLE
Add POST JSON body request support

### PR DESCRIPTION
**Pull request by antonio-garcia-sonarsource**
There is currently no way of using the request class to send a JSON in the body of the request. This is needed for SC-2114 as a new endpoint requires it.

**Checklist**

* [ ] Functional validation
* [ ] Documentation update
* [x] Update changelog

### CODE REVIEW

File: helpers/__tests__/request-test.ts [hunk 1]
```diff
@@ -146,6 +147,26 @@ describe('postJSON', () => {
...
+
+    expect(window.fetch).toBeCalledWith(url, expect.objectContaining({ method: 'POST' }));
+    expect(response.json).toBeCalled();
+  });
+
+  it('should post with a body and get json', () => {
+    postJSONBody(url, { data: 'test' });
+    expect(window.fetch).toBeCalledWith(
+      url,
+      expect.objectContaining({ body: '{"data":"test"}', method: 'POST' })
```
Comment (gregaubert) [inline helpers/__tests__/request-test.ts]: The content-type header could be checked too

File: helpers/__tests__/request-test.ts [hunk 2]
```diff
@@ -146,6 +147,26 @@ describe('postJSON', () => {
...
+    window.fetch = jest.fn().mockResolvedValue(response);
+    postJSONBody(url);
+    await new Promise(setImmediate);
+
+    expect(window.fetch).toBeCalledWith(url, expect.objectContaining({ method: 'POST' }));
+    expect(response.json).toBeCalled();
+  });
+
+  it('should post with a body and get json', () => {
+    postJSONBody(url, { data: 'test' });
```
Comment (gregaubert) [inline helpers/__tests__/request-test.ts]: Could we test with a bit more complex object to make sure it's correctly transformed ? maybe something like `{ nested: { data: 'test', withArray: [1,2] } }`

**Review by gregaubert — APPROVED** [review]
Just a small update to the tests and seems good to me!

I also added @philippe-perrin-sonarsource for a review from a SQ team member, let's wait for his input before merging.

**Review by philippe-perrin-sonarsource — APPROVED** [review]
LGTM. Although `postJSON` & `postJSONBody` namings starts to confuse me a bit.

**Pull request merged**